plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/DataflowDependencyManager.java

```diff
@@ -125,7 +125,23 @@ public boolean hasTrackedDataflowDependency(IProject project) {
         || Artifact.RELEASE_VERSION.equals(dependency.getVersion()));
   }
 
+  /**
+   * Returns a possible version range for the given project.
+   * 
+   * @throws IllegalStateException if the version range is invalid
```

Comment:
I think the Javadoc should be something callers can understand. Unclear to me as a caller when a range is invalid and how to avoid it.

Reply:
Looking at the code a bit, I think `IllegalStateException` should never be thrown, unless our code is buggy. It's when, e.g., doing `.createFromVersionSpec("nonsensical string")`, but we are sane to always pass a well-formed version string.

Reply:
This code pulls out the version specified in the user's `pom.xml` for the Dataflow Java SDK (`com.google.cloud.dataflow:google-cloud-dataflow-java-sdk-all`), and it's possible the user has invalid values.  I'll update the javadoc to make this more explicit.